ext/mbstring/libmbfl/mbfl/mbfl_encoding.c

```diff
@@ -349,7 +353,7 @@ const mbfl_encoding *mbfl_name2encoding_ex(const char *name, size_t name_len)
 	/* search MIME charset name */
 	for (encoding = mbfl_encoding_ptr_list; *encoding; encoding++) {
 		if ((*encoding)->mime_name) {
-			if (strcasecmp((*encoding)->mime_name, name) == 0) {
+			if (strcasecmp((*encoding)->mime_name, name) == 0 && strlen((*encoding)->mime_name) == name_len) {
```

Comment:
This should use strncasecmp and the same `== '\0'` check that I suggested below.